onnxruntime/core/optimizer/constant_sharing.cc

```diff
@@ -233,24 +232,17 @@ Status ConstantSharing::ApplyImpl(Graph& graph, bool& modified, int /*graph_leve
     size_t value_id = GetOrAddValueInConstantStore(std::move(init_value), const_value_store, data_store_key);
 
     // Construct a string by data type, value, and rank. Used as a key in pattern_key_to_shared_arg_map.
-    const std::string pattern_key = MakeString(SHARED_INITIALIZER_PREFIX, data_store_key, "_", value_id);
```

Comment:
MakeString shows up on the profiler. It makes use of `std::stringstring` which is slow. I think it must be possible to precompute the required length, reserve it and then concat the strings.

Reply:
did not get the point, do you mind sharing more details for how?